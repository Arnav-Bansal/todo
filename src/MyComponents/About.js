import React from 'react'
import styled from "styled-components"

const StyledLogin = styled.div`
  display: flex;
  align-items: center;
  flex-flow: column;
  width: 300px;
  height: 300px;
  margin: 40px;
  border: 2px solid #000;
  border-radius: 20px;
  background: #eee;
  h2 {
    font-family: Arial, Helvetica, sans-serif;
    font-size: 20px;
    padding: 10px
  }
  button {
    background: green;
    color: #fff;
    padding: 10px;
    margin: 5px;
    width: 150px;
    border: none;
    border-radius: 10px;
    box-sizing: border-box;
  }
`;

function ShowOwnerEducationDetails(props) {
    return (
        <StyledLogin>
        <h2>Name: {props.name}</h2>
        <h2>College: {props.college}</h2>
        <h2>Year: {props.year}</h2>
        <h2>Branch: {props.branch}</h2>
        
      </StyledLogin>
    )
    
}

function ShowOwnerBasicDetails(props) {
    return (
        <StyledLogin>
        <h2>Name: {props.name}</h2>
        <h2>Age: {props.age}</h2>
        <h1>Socials:</h1>
        <h2><a href={props.linkedin}>LinkedIn</a></h2>
        <h2><a href={props.github}>GitHub</a></h2>
        </StyledLogin>
    )
}

function getOwnerData() {
    return {
        name: "Arnav Bansal",
        age: 21,
        designation: "Developer",
        college: "IIT Ropar",
        year: "4th Year",
        branch: "CSE",
        linkedin: "https://www.linkedin.com/in/arnav-bansal-0211/",
        github: "https://github.com/Arnav-Bansal",
        gender: "Male",
        Hobbies: "Cricket, Books"
    }
}

const HOC = (Component) => {
    let props = getOwnerData()
    return function WrappedComponent() {
      return (
         <>
            <Component {...props} />
         </>
       );
    };
  };

var OwnerBasicDetails = HOC(ShowOwnerEducationDetails);
var OwnerSalaryDetails = HOC(ShowOwnerBasicDetails);

export const About = () => {
    return (
        <div>
            <h1> This is a Todo App for daily use. </h1>
            <br></br>
            <h2>Owner College Details:</h2>
	        <OwnerBasicDetails></OwnerBasicDetails><br></br>
            <h2>Owner Basic Details:</h2>
            <OwnerSalaryDetails></OwnerSalaryDetails>
        </div>
    )
}

