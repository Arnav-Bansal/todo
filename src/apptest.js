export class apps {
    constructor()
    {
        this.todos=[]
    }
    addTodo = (title, desc) => {
        console.log("I am adding this todo", title, desc)
        let sno=1
       /* if (this.todos == undefined) {
            sno = 0;
        }
        else {
            sno = this.todos[this.todos.length - 1].sno + 1;
        }*/
        const myTodo = {
            sno: sno,
            title: title,
            desc: desc,
        }
        console.log(myTodo);
        this.todos.push(myTodo)
    }

    onDelete = (todo) => {
        this.todos = this.todos.filter((e) => {
            return e == todo;
        });
    }
}
