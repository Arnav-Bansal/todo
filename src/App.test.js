import React from 'react'
import { render, screen, cleanup } from '@testing-library/react';
import App from './App';
import { onDelete } from './apptest';
import { addTodo } from './apptest';
import {apps} from './apptest';
import { AddTodo } from './MyComponents/AddTodo';

const fakeLocalStorage = (function () {         //mocking
  let store = {};

  return {
    getItem: function (key) {
      return store[key] || null;
    },
    setItem: function (key, value) {
      store[key] = value.toString();
    },
    removeItem: function (key) {
      delete store[key];
    },
    clear: function () {
      store = {};
    }
  };
})();

let todo1 = new apps
export function saveToStorage(value) {
  window.localStorage.setItem('the-key', value);
}
let todos = todo1.todos

test('renders learn react link', () => {
  render(<App />);
  const linkElement1 = screen.getByText(/About/i);
  const linkElement2 = screen.getByText(/Add Todo/i);
  const linkElement3 = screen.getByText(/My Todos List/i);

  expect(linkElement1).toBeInTheDocument();
  expect(linkElement2).toBeInTheDocument();
  expect(linkElement3).toBeInTheDocument();
});

afterEach(cleanup)

it('should take a snapshot', () => {
  const { asFragment } = render(<App />)

  expect(asFragment(<App />)).toMatchSnapshot()
});

it('addtodo func', () => {
  todo1.addTodo('Test title', 'Test desc')
  let todo = {
    sno: 1,
    title: 'Test title',
    desc: 'Test desc'
  }
  let temp = todo1.todos.filter((e) => {
    return e == todo
  });
  console.log(todo1.todos)
  expect(todo1.todos.length).toBe(1);
})

it('delete todo', () => {
  let todo = {
    sno: 1,
    title: 'Test title',
    desc: 'Test desc'
  }
  todo1.onDelete(todo)
  let temp={}
  if (todos == null) {
  }
  else {
    let temp = todo1.todos.filter((e) => {
      return e == todo
    });
  }
  expect(todo1.todos.length).toBe(0);
})


describe('storage', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'localStorage', {
      value: fakeLocalStorage,
    });
  });

  it('saves the key to the storage', () => {
    saveToStorage('fake-value');

    expect(window.localStorage.getItem('the-key')).toEqual('fake-value');
  });
});