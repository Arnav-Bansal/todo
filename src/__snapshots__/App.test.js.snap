// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        My Todos List
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          class="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/about"
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    class="container my-3"
  >
    <h3>
      Add a Todo
    </h3>
    <form>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="title"
        >
          Todo Title
        </label>
        <input
          aria-describedby="emailHelp"
          class="form-control"
          id="title"
          type="text"
          value=""
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="desc"
        >
          Todo Description
        </label>
        <input
          class="form-control"
          id="desc"
          type="text"
          value=""
        />
      </div>
      <button
        class="btn btn-sm btn-success"
        type="submit"
      >
        Add Todo
      </button>
    </form>
  </div>
  <div
    class="container"
    style="min-height: 70vh; margin: 40px auto;"
  >
    <h3
      class="my-3"
    >
      Todos List
    </h3>
    No Todos to display
  </div>
  <footer
    class="bg-dark text-light py-3"
  >
    <p
      class="text-center"
    >
      Copyright Â© MyTodosList.com
    </p>
  </footer>
</DocumentFragment>
`;
